From 45860a775736ee18ebb791a1431392ea18324f24 Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Fri, 17 Mar 2017 13:24:16 +0100
Subject: [PATCH] tc: flower: Update dest UDP port documentation

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1422629
Upstream Status: iproute2.git commit a5ae170ed87c6

commit a5ae170ed87c6a096451c7a7b94950e7befd8878
Author: Simon Horman <simon.horman@netronome.com>
Date:   Wed Jan 4 12:02:18 2017 +0100

    tc: flower: Update dest UDP port documentation

    Since 41aa17ff4668 ("tc/cls_flower: Add dest UDP port to tunnel params")
    tc flower supports setting the dest UDP port.

    * Use "port_number" to be consistent with other man-page text
    * Re-add "enc_dst_port" documentation to manpage which was
      accidently removed by b2a1f740aa4d ("tc: flower: document that *_ip
      parameters take a PREFIX as an argument.")

    Cc: Hadar Hen Zion <hadarh@mellanox.com>
    Signed-off-by: Simon Horman <simon.horman@netronome.com>
---
 man/man8/tc-flower.8 | 4 +++-
 tc/f_flower.c        | 2 +-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/man/man8/tc-flower.8 b/man/man8/tc-flower.8
index 3310496..5904a9e 100644
--- a/man/man8/tc-flower.8
+++ b/man/man8/tc-flower.8
@@ -40,7 +40,7 @@ flower \- flow based traffic control filter
 .BR enc_dst_ip " | " enc_src_ip " } { "
 .IR ipv4_address " | " ipv6_address " } | "
 .B enc_dst_port
-.IR UDP-PORT " | "
+.IR port_number
 .SH DESCRIPTION
 The
 .B flower
@@ -136,6 +136,8 @@ which have to be specified in beforehand.
 .BI enc_dst_ip " PREFIX"
 .TQ
 .BI enc_src_ip " PREFIX"
+.TQ
+.BI enc_dst_port " NUMBER"
 Match on IP tunnel metadata. Key id
 .I NUMBER
 is a 32 bit tunnel key id (e.g. VNI for VXLAN tunnel).
diff --git a/tc/f_flower.c b/tc/f_flower.c
index 41d4da7..0623ab6 100644
--- a/tc/f_flower.c
+++ b/tc/f_flower.c
@@ -58,7 +58,7 @@ static void explain(void)
 		"                       enc_src_ip [ IPV4-ADDR | IPV6-ADDR ] |\n"
 		"                       enc_key_id [ KEY-ID ] |\n"
 		"                       matching_flags MATCHING-FLAGS | \n"
-		"                       enc_dst_port [ UDP-PORT ] }\n"
+		"                       enc_dst_port [ port_number ] }\n"
 		"       FILTERID := X:Y:Z\n"
 		"       MASKED_LLADDR := { LLADDR | LLADDR/MASK | LLADDR/BITS }\n"
 		"       ACTION-SPEC := ... look at individual actions\n"
-- 
1.8.3.1

